### ISSUE TITLE
Link to open a file in workspace

### ISSUE THREAD

**Issue by bpmct** [microsoft/codetour#117]
This may be more of a VS Code question than a CodeTour question, but is it possible to link to a file in the workspace without linking to a specific tour item.

Things I tried in the tour description:
- `[./CONTRIBUTING.md](command:vscode.open?[\"${workspaceRoot}/CONTRIBUTING.md\"])`
- `[./CONTRIBUTING.md](command:vscode.open?[\"CONTRIBUTING.md\"])`
- `[./CONTRIBUTING.md](command:vscode.open?[\"CONTRIBUTING.md\"])`
- `[./CONTRIBUTING.md](command:vscode.open?[\"./CONTRIBUTING.md\"])`

Some other sites I found:
- https://stackoverflow.com/questions/43365124/get-relative-path-of-the-file-for-tasks-in-vscode
- https://github.com/microsoft/vscode/issues/53279

**Comment by gsmile** [comment]
That's a great question! @lostintangent do you have any ideas about how this could be done?

**Comment by lostintangent** [comment]
I'm actually not sure 😁 I'll do some investigation, and since this is such a common-scenario, we should probably add a shortcut for enabling this. Stay tuned!

**Comment by albertored** [comment]
According to [this](https://code.visualstudio.com/api/extension-guides/command#command-uris), I think the problem is that commands and sometimes arguments should be URI encoded.

I tried as a proof of concept to replace the [this code](https://github.com/microsoft/codetour/blob/441e9a558cbbbf0e24dca5900ee7b16f80b3ab45/src/player/index.ts#L62) with
```js
.replace(COMMAND_PATTERN, (_, commandPrefix, params) => {
      let args = JSON.parse(params)
      commandPrefix = commandPrefix.slice(1);
      if (commandPrefix == "command:vscode.open?") {
        args[0] = Uri.parse(args[0])
      }
      args = encodeURIComponent(JSON.stringify(args));
      const commandUri = Uri.parse(`${commandPrefix}${args}`);
      console.log(`(${commandUri}`)
      return `(${commandUri}`;
    })
```
and it works.

Clearly this solution is very ugly, we can not add an if clause for every possible command but I posted anyway because it can be useful for implementig the feature accordingly.

**Comment by lostintangent** [comment]
@albertored Ah great find! Thanks so much for sharing this.

I wonder if the best solution is for CodeTour to look for relative URLs and convert them into the appropriate `vscode.open` command? That way, we solve a common case, without needing to try to make the command handling too "smart"?

**Comment by albertored** [comment]
Hi @lostintangent I just double-checked and there is no need of calling `Uri.parse` on the file argument so the code can be simplified and generalized to 

```js
.replace(COMMAND_PATTERN, (_, commandPrefix, params) => {
      const args = encodeURIComponent(JSON.stringify(JSON.parse(params)));
      // the regex captures also the initial '('
      commandPrefix = commandPrefix.slice(1);
      const commandUri = Uri.parse(`${commandPrefix}${args}`);
      return `(${commandUri}`;
    })
```

I tried only with `vscode.open` command but it think it should work with any command.

Anyway it would be very useful to also have some shortcut for adding a link to a file.

**Comment by albertored** [comment]
Yes, it is a good idea

**Comment by albertored** [comment]
Also because I made it work by inserting the absolute path of the file to open, we should also add some logic to prepend the workspace root directory if the path is not absolute

**Comment by lostintangent** [comment]
I just shipped support for markdown link that reference relative file paths: `[Open this file](./file.html)`. That will translated into the appropriate link, and doesn't require the use of the `command:*` syntax.

I'll look into improving the command-syntax as follow-up, but for the purpose of this bug, the new file reference support should solve the problem, in a simpler way. Let me know if you run into any issues/feedback. Thanks!

**Comment by bpmct** [comment]
Is this resolved in the latest version? I have tried these things without success

[link](command:vscode.open?[\"${workspaceRoot}/README.md\"])
[link](./README.md)
[link](../README.md) - from one directory in

**Comment by lostintangent** [comment]
Yep this should be resolved. Note that the file path needs to be relative to the root of the workspace, not to the tour file. Did that not work for you?

**Comment by bpmct** [comment]
Yes I am still seeing this behavior

https://user-images.githubusercontent.com/22407953/116791864-1c7c0d00-aa8b-11eb-91e3-75e2b8d0965a.mov

Perhaps this is related to this?

> we should also add some logic to prepend the workspace root directory if the path is not absolute

**Comment by lostintangent** [comment]
Hmm. I updated the extension to pretend the workspace root to the relative path. So that shouldn't be an issue. I was also going to mention that the file path is case-sensitive, but you seem to be using that already.

Quick question: what OS are you testing on? I was trying to repro it in Mac and Linux, and so I may need to ensure this isn't broken on Windows.

**Comment by bpmct** [comment]
I was able to isolate the bug a bit further. It appears the link does not work if the tour is being created/edited, but if it's being played through, the links work fine:

https://user-images.githubusercontent.com/22407953/116838157-648a5500-ab9b-11eb-9c2c-0feed0196ddd.mov

**Comment by bradennapier** [comment]
Would be nice if you could "open to the side" so-as not to disrupt the ongoing tour
